### DISCUSSION TITLE
How to get the values from the Hx-Trigger header and use them in hx-include ?

### DISCUSSION THREAD

**Discussion by EdwinBekaert** [bigskysoftware/htmx#1534] [Q&A] [answered]
![image](https://github.com/bigskysoftware/htmx/assets/48248357/0b591934-1496-46aa-ab46-bd91903f3b42)

![image](https://github.com/bigskysoftware/htmx/assets/48248357/75657032-7e1a-4038-a44e-5874e2319900)

This javascript can grab the details, so how to use that in hx-include ? Or is there another way?
```
document.body.addEventListener("ProductCategoryChangedEvent", function(evt){ 
        alert(evt.detail.CategoryId); 
})
```

**Comment by andryyy** [comment, marked as answer]
Writing from the toilette on vacation, so no guarantees:

`hx-vals='js:{category: event.detail.CategoryId}'`

**Reply by andryyy** [reply]
I can’t remember the exact behavior, but I think you need to define any other parameter you want to submit  explicitly with hx-params or include it in hx-vals. Not sure if hx-include combines with it.

**Reply by EdwinBekaert** [reply]
Thanx. thats great... have a nice vacation !

ps: hx-include won't take that code
